operators/multiclusterobservability/pkg/util/managedclusteraddon.go

```diff
@@ -69,30 +50,75 @@ func CreateManagedClusterAddonCR(c client.Client, namespace string) error {
 	); err != nil && errors.IsNotFound(err) {
 		// create new managedClusterAddon
 		if err := c.Create(context.TODO(), newManagedClusterAddon); err != nil {
-			log.Error(err, "Cannot create observability-controller  ManagedClusterAddOn")
+			log.Error(err, "failed to create managedclusteraddon", "name", ManagedClusterAddonName, "namespace", namespace)
+			return err
+		}
+
+		time.Sleep(2 * time.Second)
```

Comment:
Can you add a loop to get the applied managedclusteraddon? e.g.: try 10 times per 1 sec. I am afraid it can be wrong if the environment is slow so that cannot get the applied managedclusteraddon in 2 seconds.

Reply:
done, retry with 2s interval and 10s timeout.